.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,49 @@
+name: Deploy MkDocs to GitHub Pages
+
+on:
+  push:
+    branches: [ "main" ]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r ./docs/manual/requirements.txt
+
+      - name: Build site (config in docs/manual/, output to repo root ./site)
+        run: |
+          python -m mkdocs build -f docs/manual/mkdocs.yml -d ../../site
+
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: site
```

Comment:
The material for mkdocs theme we use offers a simple tool `mike` for this entire script:

```
mike deploy --push --title "4.4" --alias-type=copy --update-aliases 4.4 latest
```

See reference: https://github.com/geonetwork/core-geonetwork/blob/main/.github/workflows/docs.yml